plugins/modules/aci_l4l7_device_selection_policy.py

```diff
@@ -0,0 +1,332 @@
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        contract=dict(type="str", aliases=["contract_name"]),
+        graph=dict(type="str", aliases=["service_graph", "service_graph_name"]),
+        node=dict(type="str", aliases=["node_name"]),
+        device=dict(type="str"),
+        context=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "contract", "graph", "node"]],
+            ["state", "present", ["tenant", "contract", "graph", "node"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    state = module.params.get("state")
+    contract = module.params.get("contract")
+    graph = module.params.get("graph")
+    node = module.params.get("node")
+    device = module.params.get("device")
+    context = module.params.get("context")
+
+    policy_dn = "ldevCtx-c-{0}-g-{1}-n-{2}".format(contract, graph, node) if (contract, graph, node) != (None, None, None) else None
```

Comment:
is there a reason that in the other module you call it ldev_ctx_rn?
should we create a re-usable constant for this?

Reply:
I didn't create a constant but used ldev_ctx_rn instead of policy_dn here